File: Sort_array_by_parity_submissions.py
class Solution(object):
    def sortArrayByParity(self, A):
        """
        :type A: List[int]
        :rtype: List[int]
        """
        A_even = []
        A_odd = []
        for a in A:
            if a % 2 == 0:
                A_even.append(a)
            else:
                A_odd.append(a)
        return A_even + A_odd

File: Sort_array_by_parity_submissions.cpp
class Solution {
public:
    vector<int> sortArrayByParity(vector<int>& A) {

        // Initialize a vectory where we'll store our sorted list
        vector<int> sorted_list;

        // Check to see if A is empty
        if (A.size() == 0){
            return A;
        }

        // Otherwise, add even elements to front of vector and odd to the back
        else{

            for(int i = 0; i < A.size(); i++){

                // Add even elements to front of the vector
                if (A[i] % 2 == 0){

                    sorted_list.insert(sorted_list.begin(), A[i]);

                }

                // Add odd elements to back of the vector
                else{

                    sorted_list.push_back(A[i]);

                }

            }

            return sorted_list;
        }

    }
};
